apps/astarte_trigger_engine/lib/astarte_trigger_engine/cqex_nodes.ex

```diff
@@ -0,0 +1,58 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.TriggerEngine.CQExNodes do
+  use Skogsra.Type
+
+  @default_port 9042
+
+  @impl Skogsra.Type
+  def cast(value)
+
+  def cast(value) when is_binary(value) do
+    value
+    |> String.split(",", trim: true)
+    |> Enum.map(fn host_port_str ->
+      [host, port] =
+        case String.split(host_port_str, ":") do
+          [host, port_str] ->
+            port =
+              with {port, _} <- Integer.parse(port_str) do
+                port
+              else
+                :error ->
+                  @default_port
+              end
+
+            [host, port]
+
+          [host] ->
+            [host, @default_port]
+
+          _ ->
+            raise("Error parsing host:port. Check your configuration variables.")
```

Comment:
You shouldn't raise here, it should return `{:ok, value}` or `:error`.
The whole function should probably be:
```elixir
value
|> String.split(",", trim: true)
|> Enum.reduce_while({:ok, []}, fn host_port_str, {:ok, acc} ->
  with [host, port_str] <- String.split(host_port_str, ":", parts: 2),
       {port, ""} <- Integer.parse(port_str) do
    {:cont, {:ok, [{host, port} | acc]}}
  else
    [host] ->
      {:cont, {:ok, [{host, @default_port} | acc]}}

    _ ->
      {:halt, :error}
  end
end)
```

Reply:
And also add an explicit function head to handle an empty string
```
def cast(""), do: :error
```